### PR TITLE
Update NavigationPathManageable.swift

### PR DESCRIPTION
**Pull request by ahmed-yamany**
```swift
func popTo(_ destination: Destination)
```

### CODE REVIEW

File: Sources/Routing/Protocols/NavigationPathManageable.swift [hunk 1]
```diff
@@ -10,28 +10,27 @@ import SwiftUI
 public protocol NavigationPathManageable: ObservableObject {
     associatedtype Destination: ViewDisplayable
 
-    var path: NavigationPath { get set }
+    var path: [Destination] { get set }
 
-    func pop(_ count: Int)
+    func popTo(_ destination: Destination)
```
Comment (JamesSedlacek) [inline Sources/Routing/Protocols/NavigationPathManageable.swift:15]:
```suggestion
    func pop(to destination: Destination)
```

Comment (JamesSedlacek) [reply Sources/Routing/Protocols/NavigationPathManageable.swift:15]: I think it'd still be useful to have a function for `func pop(by count: Int)`

File: Sources/Routing/Protocols/NavigationPathManageable.swift [hunk 2]
```diff
@@ -10,28 +10,27 @@ import SwiftUI
...
-    /// Removes the specified number of destinations from the navigation path
-    /// - Parameter count: The number of views to be popped from the navigation path
-    public func pop(_ count: Int = 1) {
-        guard count <= path.count else {
-            path = .init()
-            return
-        }
-        path.removeLast(count)
+    // Function to pop the navigation stack up to a specified destination
+    public func popTo(_ destination: Destination) {
```
Comment (JamesSedlacek) [inline Sources/Routing/Protocols/NavigationPathManageable.swift:23]: how will this handle if the destination exists multiple times in the stack?

File: Sources/Routing/Protocols/NavigationPathManageable.swift [hunk 3]
```diff
@@ -10,28 +10,27 @@ import SwiftUI
 public protocol NavigationPathManageable: ObservableObject {
     associatedtype Destination: ViewDisplayable
 
-    var path: NavigationPath { get set }
+    var path: [Destination] { get set }
```
Comment (JamesSedlacek) [inline Sources/Routing/Protocols/NavigationPathManageable.swift:13]:
```suggestion
    var stack: [Destination] { get set }
```

Would it make sense to rename this if it's no longer a NavigationPath?

File: Sources/Routing/Protocols/NavigationPathManageable.swift [hunk 4]
```diff
@@ -10,28 +10,27 @@ import SwiftUI
 public protocol NavigationPathManageable: ObservableObject {
```
Comment (JamesSedlacek) [inline Sources/Routing/Protocols/NavigationPathManageable.swift:10]:
```suggestion
public protocol DestinationManageable: ObservableObject {
```

Does it make sense to rename this now that it isn't managing a navigation path?

File: Sources/Routing/Protocols/NavigationPathManageable.swift [hunk 5]
```diff
@@ -10,28 +10,27 @@ import SwiftUI
...
+    // Function to pop the navigation stack up to a specified destination
+    public func popTo(_ destination: Destination) {
+         // Check if the destination exists in the stack
+          if let index = path.lastIndex(where: { $0 == destination }) {
+              // Remove destinations above the specified destination
+              path = Array(path.prefix(upTo: index + 1))
+          }
     }
 
     /// Resets the navigation path to its initial state
```
Comment (JamesSedlacek) [inline Sources/Routing/Protocols/NavigationPathManageable.swift:31]: Should we update these doc comments now that it isn't a navigation path?

File: Sources/Routing/Protocols/NavigationPathManageable.swift (hunk 2 above)
Comment (ahmed-yamany) [reply Sources/Routing/Protocols/NavigationPathManageable.swift:23]: this will pop to the last specified destination in the stack

Comment (JamesSedlacek) [reply Sources/Routing/Protocols/NavigationPathManageable.swift:23]: Can you write tests for this & other cases? (e.g. if it doesn't exist. if it's on top of the stack)

File: Sources/Routing/Protocols/NavigationPathManageable.swift (hunk 1 above)
Comment (ahmed-yamany) [reply Sources/Routing/Protocols/NavigationPathManageable.swift]: I returned it back

File: Sources/Routing/Protocols/NavigationPathManageable.swift [hunk 6]
```diff
@@ -28,10 +29,19 @@ extension NavigationPathManageable {
         }
         path.removeLast(count)
     }
+    
+    // Function to pop the navigation stack up to a specified destination
+    public func pop(to destination: Destination) {
+         // Check if the destination exists in the stack
+          if let index = path.lastIndex(where: { $0 == destination }) {
+              // Remove destinations above the specified destination
+              path = Array(path.prefix(upTo: index + 1))
```
Comment (JamesSedlacek) [inline Sources/Routing/Protocols/NavigationPathManageable.swift:38]: We could make this more readable if we had an array extension..

Maybe something like this?
```swift
extension Array {
    /// Truncates the array up to the specified index.
    ///
    /// This method mutates the original array, keeping the elements up to the given index.
    /// If the index is greater than the array's count, the array remains unchanged.
    ///
    /// - Parameter index: The index up to which the array should be truncated.
    ///
    /// Example Usage:
    /// ```
    /// var numbers = [1, 2, 3, 4, 5]
    /// numbers.truncate(to: 2)
    /// // numbers is now [1, 2, 3]
    /// ```
    mutating func truncate(to index: Int) {
        guard index < self.count else { return }
        self = Array(self[..<Swift.min(index + 1, self.count)])
    }
}
```

File: Sources/Routing/Protocols/NavigationPathManageable.swift (hunk 2 above)
Comment (ahmed-yamany) [reply Sources/Routing/Protocols/NavigationPathManageable.swift]: I made a new pull request that has all the changes include unit test

**Pull request closed without merging**